constructor-manager-api/src/constructor_manager_api/utils/worker.py

```diff
@@ -0,0 +1,110 @@
+"""Constructor updater api worker."""
+
+import json
+import logging
+from typing import List
+
+from qtpy.QtCore import QObject, QProcess, Signal  # type: ignore
+
+from constructor_manager_api.utils.conda import get_prefix_by_name
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConstructorManagerWorker(QObject):
+    """A worker to run the constructor manager cli and process errors.
+
+    Parameters
+    ----------
+    args : list
+        Arguments to pass to the constructor manager.
+    detached : bool, optional
+        Run the process detached, by default ``False``.
+    """
+
+    _WORKERS: "List[ConstructorManagerWorker]" = []
+    finished = Signal(dict)
+
+    def __init__(self, args, detached=False):
+        super().__init__()
+        ConstructorManagerWorker._WORKERS.append(self)
+        self._detached = detached
+        self._program = self._executable()
+
+        if not self._program.is_file():
+            raise FileNotFoundError(f"Could not find {self._program}")
+
+        self._args = args
+        self._process = QProcess()
+        self._process.setArguments(args)
+        self._process.setProgram(str(self._program))
+        self._process.finished.connect(self._finished)
+
+    @staticmethod
+    def _executable():
+        """Get the executable for the constructor manager."""
+        bin = "constructor-manager-cli"
+        envs = ["_constructor-manager", "constructor-manager", "base"]
+        for env in envs:
+            program = get_prefix_by_name(env) / "bin" / bin
```

Comment:
Was checking on Windows and I think there instead of `bin` the directory where executables are stored is `Scripts`

Reply:
HI @dalthviz, indeed, I ran into the same issue. Thanks for checking!

Reply:
Double checking it appears in my env as an `.exe`:

![imagen](https://user-images.githubusercontent.com/16781833/225344421-ccaa4b97-44c2-4613-bee5-5ce118ea43d6.png)

I think the `.bat` executables exist but for `conda` inside the `condabin` directory:

![imagen](https://user-images.githubusercontent.com/16781833/225344319-34dabc2e-f6f0-4d74-937f-1fef845c037f.png)

Reply:
On Windows entry points use a C launcher that mimics what a shebang would do. 

`constructor-manager-cli.exe` will find:

- constructor-manager-cli-script.py` next to itself
- a `python.exe` interpreter nearby

After that it will run `python.exe constructor-manager-cli-script.py`.

So yes, it's an `.exe` in `Scripts`. I _think_ a subprocess without the `.exe` extension still finds the required executable, but I am not sure. Better be explicit.